helper/testhelpers/docker/testhelpers.go

```diff
@@ -135,6 +136,21 @@ func (d *Runner) StartService(ctx context.Context, connect ServiceAdapter) (*Ser
 	}
 
 	cleanup := func() {
+		if d.RunOptions.LogConsumer != nil {
+			rc, err := d.DockerAPI.ContainerLogs(ctx, container.ID, types.ContainerLogsOptions{
+				ShowStdout: true,
+				ShowStderr: true,
+				Timestamps: true,
+				Details:    true,
+			})
+			if err == nil {
+				b, err := ioutil.ReadAll(rc)
+				if err == nil {
+					d.RunOptions.LogConsumer(string(b))
+				}
+			}
```

Comment:
should we send the err string to the LogConsumer if err != nil?

Reply:
Good call.

Reply:
Outputting the err from DockerAPI.ContainerLogs might be useful as well, say the container id completely disappeared or we had issues with docker daemon itself, it might contain further clues?

Reply:
That'll have to wait for a future PR, I've reached my timebox limit for this effort.